### PR TITLE
Switch to cmake / Fedora 34 / Ubuntu Focal

### PR DESCRIPTION
**Pull request by morxa**
Switch the build system to cmake and upgrade to Fedora 34 and Ubuntu Focal on the way.

This replaces the build system completely.

Install targets and other non-default configurations are not (yet) supported.

### CODE REVIEW

**Review by TarikViehmann — CHANGES_REQUESTED** [review]
Looks great and works smooth as far as i can tell.

A minor thing: you forgot to remove the toplevel Makefile

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+# Read the full text in the LICENSE.md file.
+#
+
+cmake_minimum_required(VERSION 3.14)
+
+project(gazebo-rcll)
+include(GNUInstallDirs)
+
+set(USERDIR
+    ".config/rcll-refbox"
```
Comment (TarikViehmann) [inline CMakeLists.txt:25]: Is there a reason why this is named rcll-refbox?

Comment (morxa) [reply CMakeLists.txt:25]: Good catch, it should be `gazebo-rcll` instead. Or even better `.config/rcll/gazebo`?

**Review by TarikViehmann — APPROVED** [review]
Awesome work, thanks a lot!

**Comment by morxa** [comment]
Thanks for reviewing!

**Pull request merged**